src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,505 @@ title: Builder.io & Astro
+</FileTree>
+
+Then add the following content:
+
+```astro title="src/pages/builder-preview.astro"
+---
+const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;
+---
+
+<html lang="en">
+  <head>
+    <title>Preview for builder.io</title>
+  </head>
+  <body>
+    <header>This is your header</header>
+
+    <builder-component model={builderModel} api-key={builderAPIpublicKey}
+    ></builder-component>
+    <script async src="https://cdn.builder.io/js/webcomponents"></script>
+
+    <footer>This is your footer</footer>
+  </body>
+</html>
+
+```
+
+In the above example, `<builder-component>` tells Builder where to insert the content from its CMS.
+
+#### Setting the new route as the preview URL
+
+1. Copy the full URL of your preview, including the protocol, to your clipboard (e.g. `https://{your host}/builder-preview`). 
+
+2. Go to the **Models** tab in your Builder space, pick the model you've created and paste the URL from step 1 into the **Preview URL** field. Make sure the URL is complete and includes the protocol, for example `https://`.
+
+3. Click the **Save** button in the upper right.
+
+:::tip
+When you deploy your site, change the preview URL to match your production URL, for example `https://myAwesomeAstroBlog.com/builder-preview`.
+:::
+
+#### Testing the preview URL setup
+
+1. Make sure your site is live (e.g. your dev server is running) and the `/builder-preview` route is working.
+
+2. In your Builder space under the **Content** tab, click on **New** to create a new content entry for your `blogpost` model.
+
+3. In the Builder editor that just opened, you should be able to see the `builder-preview.astro` page with a big **Add Block** in the middle.
+
+:::tip[Troubleshooting]
+
+Things can sometimes go wrong when setting up the preview. If something's not right, you can try one of these things:
+
+* Make sure the site is live - for example, your dev server is running.
+* Make sure that the URLs match exactly - the one in your Astro project and the one set in the Builder app.
+* Make sure it's the full URL including the protocol, for example `https://`.
+* If you're working in a virtual environment like [StackBlitz](https://stackblitz.com/) or [Gitpod](https://www.gitpod.io/), you might have to copy and paste the URL again when you restart your workspace, since this usually generates a new URL for your project.
+
+For more ideas, read [Builder's troubleshooting guide](https://www.builder.io/c/docs/guides/preview-url-working).
+:::
+
+### Creating a blog post
+
+1. Before you can create new content in Builder, make sure [the preview URL for your model is working](#setting-up-the-preview).
+2. Open the visual editor for a content entry of our `blogpost` model. If you've followed [the previous section](#testing-the-preview-url-set-up), you can use that entry. Otherwise, create a new one: Go to the **Content** tab in your Builder space and click on **New**.
```

Comment:
```suggestion
2. Open the visual editor for a content entry of our `blogpost` model. If you've followed [the previous section](#testing-the-preview-url-setup), you can use that entry. Otherwise, create a new one: Go to the **Content** tab in your Builder space and click on **New**.
```
fixing link to mach changed header text